p2p/src/main/kotlin/maru/p2p/messages/StatusMessageSerDe.kt

```diff
@@ -8,16 +8,42 @@
  */
 package maru.p2p.messages
 
+import maru.compression.MaruCompressor
 import maru.p2p.Message
 import maru.p2p.RpcMessageType
 import maru.p2p.Version
-import maru.serialization.SerDe
+import maru.serialization.compression.MaruNoOpCompressor
+import maru.serialization.rlp.MaruCompressorRLPSerDe
+import maru.serialization.rlp.RLPSerDe
+import org.hyperledger.besu.ethereum.rlp.RLPInput
+import org.hyperledger.besu.ethereum.rlp.RLPOutput
 
 class StatusMessageSerDe(
-  private val statusSerDe: SerDe<Status>,
-) : SerDe<Message<Status, RpcMessageType>> {
-  override fun serialize(value: Message<Status, RpcMessageType>): ByteArray = statusSerDe.serialize(value.payload)
+  private val statusSerDe: RLPSerDe<Status>,
+  compressor: MaruCompressor = MaruNoOpCompressor(),
```

Comment:
Here we use `MaruNoOpCompressor` because the compressed size for Status message is going to be larger than the raw size (i.e. few bytes larger), we might even consider just to extend `RLPSerDe` instead of `MaruCompressorRLPSerDe`